### DISCUSSION TITLE
IDwarp issue related to defining boundary conditions

### DISCUSSION THREAD

**Discussion by BilalAnwar165** [mdolab/dafoam#689] [General]
Hi

Can anyone pls help me on this? I am stuck on this for the past 2 weeks and is really frustrated now.

I am using IDWarp to deform a volume mesh. I have created a fluid domain with an Onera wing inside it. The unstructured mesh (Tri, tetra) is created using ICEM and exported in .cgns format.

When I read this mesh using IDWarp it gives me th following error.

"IDWarp cannot handle boundary conditions defined by "PointRange" or "PointList". Please use boundary conditions defined by "ElementRange" or "ElementList" instead"

I don't have any idea how to do define that. I have tried different strategies to give BC inside ICEM. but none is working.

Any help on this appreciated. I am attaching the screen shot of the mesh, boundary conditions and the error. @friedenhe can you help me on this?

![error](https://github.com/user-attachments/assets/7bc20fc9-a189-40cb-99cf-3b35ffabee5e)
![boundary conditions](https://github.com/user-attachments/assets/ad60aa22-8dae-4abf-bff7-dbf58e5ba621)
![mesh](https://github.com/user-attachments/assets/dd9e978a-9e44-41c9-ac2d-3501545928c1)

**Comment by friedenhe** [comment]
It seems that you are trying to use IDWarp as a stand-alone tool to deform volume meshes. Have you tried this example from IDWarp? https://github.com/mdolab/idwarp/blob/main/examples/unstructured/mdo_tutorial_ustruct.py Please check the BC setup in `input_files/mdo_tutorial_face_bcs.cgns` (you need to first run input_files/get-input-files.sh to download the cgns)

**Reply by BilalAnwar165** [reply]
Yes I am trying to run IDWarp as a stand-alone tool. I have already run all the input files, downloaded from the given .sh file. They run fine. I even created my BC similar to the given mesh for onera wing. but it's outputting the same error. Please help me with this as I am getting frustrated now. I have to warp an unstructured mesh over an onera wing (tri/tetra elements).
I have attached the ss for the BC used in the _**/mdo_tutorial_face_bcs.cgns**_ file. You can see that the BC I am using and the BC shown in this file are similar.

![bc](https://github.com/user-attachments/assets/a436590d-0b10-460a-af65-8c01b7931a98)

**Reply by BilalAnwar165** [reply]
I am attaching my mesh file as well for your perusal @friedenhe . It is created in ICEM 19 and exported as .cgns with 3.2ADF version.

[onera10.zip](https://github.com/user-attachments/files/17143117/onera10.zip)

**Reply by BilalAnwar165** [reply]
Secondly, I will like to know in which software/tool this mesh has been created (mdo_tutorial_face_bcs.cgns)! Even if this mesh is opened in ICEM and then saved as .cgns again, it does not work with IDwarp. Is there any specific format of CGNS that IDWarp is using @friedenhe  @sseraj ?

**Reply by BilalAnwar165** [reply]
@friedenhe @sseraj is there anyone who can give some insight of any lead please?
